### DISCUSSION TITLE
Handling dynamic folder names, such as Firefox user.js

### DISCUSSION THREAD

**Discussion by x0ba** [twpayne/chezmoi#2905] [Q&A]
I'd like to put my firefox `user.js` file in my chezmoi dotfiles, but I have yet to find out how to handle that. The user.js file goes into a differently named folder every time, `/.mozilla/firefox/*.default-release`. Is there a way to handle those in Chezmoi?

**Comment by nandalopes** [comment]
Perhaps using a link to another, fixed named folder, and keeping the target folder name in chezmoi config file.

After that, you can use a `run_onchange` script to update the link to the correct folder.

**Comment by bradenhilton** [comment]
I have [a repo](https://github.com/bradenhilton/chezmoi-firefox-setup) outlining how to do this, but it's currently under reconstruction to incorporate more recent chezmoi features.

Basically, you can do achieve it by [adapting this `profiles.ini` template](https://github.com/twpayne/chezmoi/issues/1226#issuecomment-867228095) to your use case.

Thanks for the indirect nudge, I'll see if I can finish the repo soon.

**Comment by Lockszmith-GH** [comment]
I think your solution is actually not a chezmoi one, but a firefox one.

Firefox profiles can be anywhere you want them, and you can control that.
There is an .ini file in a pre-defined location pointing to all profiels, and then the profiles themselves.

I place my profile outside of the regular firefox direcotries, so I can do easy backups of it.

`about:profiles` will give you access to the create a new profile.
I don't remember the .ini file location, but I'm sure that's an easy google search away.

Hope this helps.

**Reply by Lockszmith-GH** [reply]
🤦 I wrote before reading, I see [bradenhilton](https://github.com/twpayne/chezmoi/discussions/2905#discussioncomment-5509228) provided much more detail on exactly this.
